gulpfile.js

```diff
@@ -6,245 +6,131 @@
 
 const gulp = require('gulp');
 const args = require('yargs').argv;
-const colors = require('colors');
 const fs = require('fs');
 const util = require('util');
 const path = require('path');
 const glob = require('glob');
 const execSync = require('child_process').execSync;
 const jsonStableStringify = require('json-stable-stringify');
 
-var mappings = require('./codegen_mappings.json');
-
+const azureSDKForNodeRepoRoot = __dirname;
 const defaultAutoRestVersion = '1.2.2';
 var usingAutoRestVersion;
-const specRoot = args['spec-root'] || 'https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification';
-const project = args['project'];
+let azureRestAPISpecsRoot = args['azure-rest-api-specs-root'] || path.resolve(azureSDKForNodeRepoRoot, '..', 'azure-rest-api-specs');
+const package = args['package'];
 const use = args['use'];
-var modeler = 'Swagger';
 const regexForExcludedServices = /\/(intune|documentdbManagement|insightsManagement|insights|search)\//i;
 
-function getAutorestVersion(version) {
-  if (!version) version = 'latest';
-  let getVersion, execHelp;
-  let result = true;
-  try {
-    let getVersionCmd = `autorest `;
-    let execHelpCmd = `autorest --help`;
-    console.log(getVersionCmd);
-    getVersion = execSync(getVersionCmd, { encoding: 'utf8' });
-    //console.debug(getVersion);
-    console.log(execHelpCmd);
-    execHelp = execSync(execHelpCmd, { encoding: 'utf8' });
-    //console.debug(execHelp);
-  } catch (err) {
-    result = false;
-    console.log(`An error occurred while getting the "${version}" of autorest and executing "autorest --help":\n ${util.inspect(err, { depth: null })}.`);
-  }
-  return result;
-}
+function findReadmeNodejsMdFilePaths(azureRestAPISpecsRoot) {
```

Comment:
Can't we use something like node-glob for this?

Reply:
Yeah, we probably can. I can look into that.

Reply:
Done.